src/reader-activation-newsletters/newsletters-modal.js

```diff
@@ -11,6 +11,45 @@ export function getModalContainer() {
 	);
 }
 
+/**
+ * Refresh the newsletters signup modal content.
+ *
+ * @param {string} email The email address to populate in the modal.
+ * @return {void}
+ */
+export function refreshNewslettersSignupModal( email ) {
+	const container = getModalContainer();
+	if ( ! container ) {
+		return;
+	}
+
+	const modal = container.closest( '.newspack-newsletters-signup-modal' );
+	if ( modal ) {
+		fetch( `/wp-json/newspack/v1/reader-newsletter-signup-lists/${ email }` )
+			.then( res => {
+				res
+						.json()
+						.then( ( { html } ) => {
+							if ( html ) {
+								// Create dom node from html.
+								const parser = new DOMParser();
+								const doc = parser.parseFromString( html, 'text/html' );
+								// Remove existing form.
+								const existingForm = container.querySelector( 'form' );
+								if ( existingForm ) {
+									existingForm.remove();
+								}
+								// Append new form.
+								const newForm = doc.querySelector( '.newspack-newsletters-signup form' );
+								if ( newForm ) {
+									container.appendChild( newForm );
```

Comment:
Just noting we can't simply replace the entire container as this causes the form to close prematurely, likely a result of some existing JS events.